src/main/java/uk/gov/hmcts/reform/idam/service/UserRoleService.java

```diff
@@ -59,4 +57,84 @@ public List<String> filterUsersWithRoles(List<String> staleUsers) {
             .toList();
 
     }
+
+    public void mergeRoleAssignments(String archivedUserId, String activeUserId) {
+        var response = roleAssignmentClient.getRoleAssignmentsByUserId(securityUtil.getAuthHeaders(), archivedUserId);
+
+        RoleAssignmentsMergeRequest mergeRequest = createMergeRequest(response, archivedUserId, activeUserId);
+        Map<String, String> headers = securityUtil.getAuthHeaders();
+        try (var mergeResponse = roleAssignmentClient.createRoleAssignment(headers, mergeRequest)) {
+            if (mergeResponse.status() >= 300) {
+                duplicateUserSummary.increaseFailedMerge();
+                log.error("Merge failed for user active user {} (deleted user id {})", activeUserId, archivedUserId);
```

Comment:
It would be nice to add MARKER for all merge request logs for easier appinsights search